torch/lib/c10d/frontend.cpp

```diff
@@ -833,4 +892,22 @@ c10::intrusive_ptr<ProcessGroup::Work> DistributedC10d::barrier(
   return empty_work;
 }
 
+void DistributedC10d::registerProcessGroupName(const c10::intrusive_ptr<ProcessGroup>& process_group, const std::string& name) {
+  auto it = std::find_if(
+      pg_names_.begin(),
+      pg_names_.end(),
+      [&](const std::pair<c10::intrusive_ptr<ProcessGroup>, std::string>&
+              pg_name) { return pg_name.second == name; });
+
+  TORCH_CHECK(it == pg_names_.end(), "Requested name already exists: ", name);
```

Comment:
I think fi the requested processGroupName already exists, we can just warn the user, and use that existed pg when needed? that way user won't need to create it again and again

Reply:
Changed it to only error out when found instance is not same as argument `process_group`